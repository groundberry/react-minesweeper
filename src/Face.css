.Face {
  margin: 30px auto 0;
  height: 50px;
  width: 50px;
}

.Face--button {
  background-color: #ccc;
  border-color: #d8d8d8;
  border-style: outset;
  border-width: medium;
  color: #000;
  text-align: center;
}

.Face--button:hover {
  background-color: #dcdcdc;
  border-color: #d0d0d0;
  border-style: outset;
  border-width: medium;
  cursor: pointer;
}

.Face--image {
  background-color: #ccc;
  background-size: 100% 100%;
}
