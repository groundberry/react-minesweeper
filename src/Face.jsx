import React from 'react';
import PropTypes from 'prop-types';
import { getStatusImage } from './utils/game';
import './Face.css';

const imgCool = require('./img/minesweeper_cool_face.png');

function Face(props) {
  const { gameOver, winGame, onClick } = props;
  let img = null;

  if (winGame) {
    img = imgCool;
  } else {
    img = getStatusImage(gameOver);
  }

  return (
    <button
      style={{ backgroundImage: `url(${img})` }}
      className="Face Face--button Face--image"
      onClick={onClick}
    />
  );
}

Face.propTypes = {
  gameOver: PropTypes.bool,
  winGame: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

Face.defaultProps = {
  gameOver: false,
  winGame: false,
};

export default Face;
