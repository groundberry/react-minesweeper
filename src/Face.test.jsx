import React from 'react';
import { shallow } from 'enzyme';
import Face from './Face';

describe('Face', () => {
  let onClick;

  beforeEach(() => {
    onClick = jest.fn();
  });

  it('renders a smiley face when not game over', () => {
    const wrapper = shallow(<Face onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a sad face when game over', () => {
    const wrapper = shallow(<Face gameOver onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the onClick callback when clicked', () => {
    const wrapper = shallow(<Face onClick={onClick} />);
    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
