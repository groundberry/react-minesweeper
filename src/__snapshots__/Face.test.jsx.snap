// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Face renders a cool face when win game 1`] = `
<button
  className="Face Face--button Face--image"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(minesweeper_cool_face.png)",
    }
  }
/>
`;

exports[`Face renders a sad face when game over 1`] = `
<button
  className="Face Face--button Face--image"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(minesweeper_sad_face.png)",
    }
  }
/>
`;

exports[`Face renders a smiley face when not game over 1`] = `
<button
  className="Face Face--button Face--image"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(minesweeper_smiley_face.png)",
    }
  }
/>
`;
